lib/compute/agent-node-config.ts

```diff
@@ -190,4 +193,61 @@ export class AgentNodeConfig {
        useEphemeralDevices: false,
      };
    }
+
+   private getMacTemplate(stack: Stack, props: AgentNodeNetworkProps): { [x: string]: any; } {
+     return {
+       ami: 'ami-0379811a08268a97e',
+       amiType:
+        { macData: { sshPort: '22' } },
+       associatePublicIp: false,
+       connectBySSHProcess: false,
+       connectionStrategy: 'PRIVATE_IP',
+       customDeviceMapping: '/dev/sda1=:300:true:gp3::encrypted',
+       deleteRootOnTermination: true,
+       description: 'jenkinsAgentNode-Jenkins-Agent-Mac-M1',
+       ebsEncryptRootVolume: 'ENCRYPTED',
+       ebsOptimized: true,
+       hostKeyVerificationStrategy: 'OFF',
+       iamInstanceProfile: this.AgentNodeInstanceProfileArn,
+       labelString: 'Jenkins-Agent-Mac-M1',
+       maxTotalUses: -1,
+       minimumNumberOfInstances: 1,
+       minimumNumberOfSpareInstances: 0,
+       mode: 'NORMAL',
```

Comment:
Please use exclusive mode not normal, so that workflows will only schedule on mac agent when specifically calling with labels.